### PR TITLE
Support for [TAG tag] on AI.SCRIPTSET、AI.MODELSET

### PR DESCRIPTION
**Pull request by dengliming**
for [https://github.com/RedisAI/redisai-go/issues/15](https://github.com/RedisAI/redisai-go/issues/15) and [https://github.com/RedisAI/redisai-go/issues/14](https://github.com/RedisAI/redisai-go/issues/14)

### CODE REVIEW

File: redisai/model.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package redisai
 
 import (
+	"github.com/RedisAI/redisai-go/redisai/implementations"
```
Comment (filipecosta90) [inline redisai/model.go:4]: wanted to avoid circular dependencies. please see the next comments

File: redisai/model.go [hunk 2]
```diff
@@ -17,58 +18,60 @@ type ModelInterface interface {
...
-	if len(inputs) > 0 {
-		args = args.Add("INPUTS").AddFlat(inputs)
-	}
-	if len(outputs) > 0 {
-		args = args.Add("OUTPUTS").AddFlat(outputs)
-	}
-	args = args.Add("BLOB")
-	args = args.Add(blob)
-	return args
+	modelInterface := implementations.NewModel(backend, device)
```
Comment (filipecosta90) [inline redisai/model.go:26]: avoid creating a structure just to retrieve redis.Args ( please see the https://github.com/dengliming/redisai-go/pull/1 )

File: redisai/model.go [hunk 3]
```diff
@@ -17,58 +18,60 @@ type ModelInterface interface {
...
-	var blob []byte
-	err, backend, device, blob = modelGetParseReply(reply)
-	if err != nil {
-		return err
-	}
-	model.SetBackend(backend)
-	model.SetDevice(device)
-	model.SetBlob(blob)
-	return
+func modelRunFlatArgs(name string, inputTensorNames, outputTensorNames []string) redis.Args {
```
Comment (filipecosta90) [inline redisai/model.go:56]: this should not work given that we're mixing modelset and modelrun args. please see the correction on https://github.com/dengliming/redisai-go/pull/1

File: redisai/model.go [hunk 4]
```diff
@@ -17,58 +18,60 @@ type ModelInterface interface {
...
+	modelInterface := implementations.NewEmptyModel()
+	modelInterface.SetInputs(inputTensorNames)
+	modelInterface.SetOutputs(outputTensorNames)
+	return modelSetInterfaceArgs(name, modelInterface)
 }
 
-func modelGetParseReply(reply interface{}) (err error, backend string, device string, blob []byte) {
+func modelGetParseToInterface(reply interface{}, model ModelInterface) (err error) {
```
Comment (filipecosta90) [inline redisai/model.go:63]: avoid creating a structure and depending on the implementations package 
please see https://github.com/dengliming/redisai-go/pull/1

File: redisai/model.go (hunk 1 above)
Comment (dengliming) [reply redisai/model.go]: Thanks. My mistake.

File: redisai/model.go (hunk 2 above)
Comment (dengliming) [reply redisai/model.go]: Thanks for helping. Merged.

File: redisai/model.go (hunk 4 above)
Comment (dengliming) [reply redisai/model.go:75]: Thanks. Merged.

**Comment by filipecosta90** [comment]
thank you @dengliming . merged =)

**Pull request merged**